p4-16/psa/psa.p4

```diff
@@ -568,12 +579,13 @@ control ComputeChecksum<H, M>(inout H hdr, inout M user_meta);
 
 control Deparser<H>(packet_out buffer, in H hdr);
 
-package PSA_Switch<H, M>(Parser<H, M> p,
-                         VerifyChecksum<H, M> vr,
+package PSA_Switch<H, M>(IngressParser<H, M> ip,
                          Ingress<H, M> ig,
+                         Deparser<H> igdep,
+                         EgressParser<H, M> ep,
                          Egress<H, M> eg,
                          ComputeChecksum<H, M> ck,
-                         Deparser<H> dep);
+                         Deparser<H> egdep);
```

Comment:
1. I'm happy making separate H and M for ingress and egress.  IH, IM, EH, EM.
2. Please pick something that you consider implementable, and please let's write that into the draft.  So far, I thought that the thinking was that deparser code was going to be restricted to nothing but emit() method calls, but I can't think of any objection to allowing checksum calculations in there as well.  Neither can I think of any objection to allowing checksum calculations in Ingress and/or Egress control blocks.